File: android/src/main/kotlin/com/borda/zebra_rfid_reader_sdk/utils/TagLocationingResponse.kt
package com.borda.zebra_rfid_reader_sdk.utils

import com.google.gson.Gson

object TagLocationingResponse {
    private var tag: String = ""
    private var distancePercent: Int? = null
    private var isAnyReaderConnected: Boolean = false


    fun setTag(tag: String) {
        this.tag = tag
    }

    fun getTag(): String {
        return tag
    }

    fun setAnyReaderConnected(isAnyReaderConnected: Boolean) {
        this.isAnyReaderConnected = isAnyReaderConnected
    }

    fun setDistancePercent(distancePercent: Int?) {
        this.distancePercent = distancePercent
    }

    fun reset() {
        this.tag = ""
        this.isAnyReaderConnected = false
        this.distancePercent = null
    }

    fun toJson(): String {
        return Gson().toJson(LocationInfo(tag, distancePercent, isAnyReaderConnected))
    }

}

File: ios/Classes/TagLocationingResponse.swift
import Foundation

class TagLocationingResponse {
    static let shared = TagLocationingResponse()
    
    private var tag: String = ""
    private var distancePercent: Int? = nil
    private var isAnyReaderConnected: Bool = false

    private init() {}
    
    func setTag(tag: String) {
        self.tag = tag
    }
    
    func getTag() -> String {
        return tag
    }
    
    func setAnyReaderConnected(isAnyReaderConnected: Bool) {
        self.isAnyReaderConnected = isAnyReaderConnected
    }
    
    func setDistancePercent(distancePercent: Int?) {
        self.distancePercent = distancePercent
    }
    
    func reset() {
        self.tag = ""
        self.isAnyReaderConnected = false
        self.distancePercent = nil
    }
    
    func toJson() -> String? {
        let locationInfo = LocationInfo(tag: tag, distanceAsPercentage: distancePercent, isAnyReaderConnected: isAnyReaderConnected)
        let jsonEncoder = JSONEncoder()
        if let jsonData = try? jsonEncoder.encode(locationInfo) {
            return String(data: jsonData, encoding: .utf8)
        }
        return nil
    }
}
